tests/test_pipelines_common.py

```diff
@@ -9,4 +18,79 @@ class PipelineTesterMixin:
     equivalence of dict and tuple outputs, etc.
     """
 
-    pass
+    def test_save_load(self):
+        device = "cpu"
+        components = self.get_common_pipeline_components()
```

Comment:
Quick note. Currently these tests will be run all the time, so even:

```
pytest tests/test_pipelines_common.py::PipelineTesterMixin
```

will be run, which it shouldn't, it should just be a plug in for other pipelines. 

What I'd maybe to is to just do the following:
```suggestion
        components = self.get_common_pipeline_components()
        if components is None:
            return
```

so that the test is only run for classes that define a get_commen_pipeline_components

Reply:
Actually maybe better to just go over `pipeline_class` e.g. 

```py
if self.pipeline_class is None:
    return
```

Reply:
`PipelineTesterMixin` is invisible to `pytest`, as it doesn't inherit from `unittest.TestCase`, so we should be fine!